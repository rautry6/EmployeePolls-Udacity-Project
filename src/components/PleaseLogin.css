.login-page{
    height: 932px;
    background-color: wheat;
    margin-top: 20px;
}

.invalid-login{
    color: red;
    font-size: 1.2rem;
    font-weight: bold;
}

.login-field{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 20px;
    size: 20in;
}

.please-login{
    font-size: 2rem;
}

.title{
    font-size: 2rem;
    font-weight: bold;
}

.login{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 80px;
}

.username{
    font-size: 20px;
}

.password{
    font-size: 20px;
}

.submit-button{
    font-size: 20px;
    font-weight: bold;
    background-color: rgb(218, 209, 209);
    color: rgb(0, 0, 0);
    border: 0.1cm solid rgb(0, 0, 0); 
    border-radius: 0.5cm;
    width: 100px;
    height: 30px;
    margin-top: 10px;
}