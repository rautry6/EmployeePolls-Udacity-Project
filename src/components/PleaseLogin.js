import { connect } from "react-redux/es/exports";
import { useState } from "react";
import { setAuthedUser } from "../actions/authedUser";
import { useNavigate } from "react-router-dom";

const PleaseLogin = (props) => {
  const nav = useNavigate();
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState(false);

  let users = props.users;

  const handleSubmit = (e) => {
    e.preventDefault();
    if (username in users) {
      if (users[username].password === password) {
        props.dispatch(setAuthedUser(users[username]));
        nav("/homepage");
      } else {
        setError(true);
        setUsername("");
        setPassword("");
      }
    } else {
      setError(true);
      setUsername("");
      setPassword("");
    }
  };

  const handleUsername = (e) => {
    setUsername(e.target.value);
  };

  const handlePassword = (e) => {
    setPassword(e.target.value);
  };

  return (
    <div className="login-page">
    <h3 >Please Login to View Page</h3>
      <h1>Employee Polls</h1>
      <h2>Login</h2>
      <div>
        {error ? (
          <div>
            <h3 itemID="invalid-login">Invalid username or password</h3>{" "}
            <p>User</p>
            <input
              type="text"
              value={username}
              onChange={handleUsername}
            />
            <p>Password</p>
            <input
              type="password"
              value={password}
              onChange={handlePassword}
            />
            <p>
              <button onClick={handleSubmit}>Login</button>
            </p>
          </div>
        ) : (
          <div>
            <p>User</p>
            <input
              type="text"
              value={username}
              onChange={handleUsername}
            />
            <p>Password</p>
            <input
              type="password"
              value={password}
              onChange={handlePassword}
            />
            <p>
              <button onClick={handleSubmit}>Login</button>
            </p>
          </div>
        )}
      </div>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    users: state.users,
  };
}

export default connect(mapStateToProps)(PleaseLogin);
