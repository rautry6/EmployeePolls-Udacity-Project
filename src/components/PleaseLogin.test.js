import { render, fireEvent } from "@testing-library/react";
import PleaseLogin from "./PleaseLogin";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../reducers";
import middleware from "../middleware";

const store = createStore(rootReducer, middleware);

describe("PleaseLogin", () => {
  it("will display failed login text if username or password is incorrect", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <PleaseLogin />
      </Provider>
    );
    const usernameInput = getByTestId("login-input");
    const passwordInput = getByTestId("password-input");
    const submitButton = getByTestId("submit-button");
    fireEvent.change(usernameInput, { target: { value: "test" } });
    fireEvent.change(passwordInput, { target: { value: "test" } });
    fireEvent.click(submitButton);
    expect(getByTestId("failed-login-text")).toBeInTheDocument();
  });
  it("will match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <PleaseLogin />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
