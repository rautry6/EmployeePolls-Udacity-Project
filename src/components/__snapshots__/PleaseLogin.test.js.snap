// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PleaseLogin will match snapshot 1`] = `
<div>
  <div
    class="please-login-page"
  >
    <h3
      class="please-login"
    >
      Please Login to View Page
    </h3>
    <h1
      class="title"
    >
      Employee Polls
    </h1>
    <h2
      class="login"
    >
      Login
    </h2>
    <div
      class="login-field"
    >
      <div>
        <p
          class="username"
        >
          Username
        </p>
        <input
          class="login-input"
          data-testid="login-input"
          type="text"
          value=""
        />
        <p
          class="password"
        >
          Password
        </p>
        <input
          class="password-input"
          data-testid="password-input"
          type="password"
          value=""
        />
        <p>
          <button
            class="submit-button"
            data-testid="submit-button"
          >
            Login
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;
